### PR TITLE
add caddy-cis plugin

### PR DESCRIPTION
**Pull request by juju2013**
### 1. What does this change do, exactly?
A middleware makes request to path (some what) case insensitive.

### 2. Please link to the relevant issues.
#569

### 3. Which documentation changes (if any) need to be made because of this PR?
None

### 4. Checklist

- [X] I have written tests and verified that they fail without my change
- [X] I have squashed any insignificant commits
- [X] This change has comments for package types, values, functions, and non-obvious lines of code
- [X] I am willing to help maintain this change if there are issues with it later

### CODE REVIEW

File: caddy/caddymain/run.go [hunk 1]
```diff
@@ -21,6 +21,8 @@ import (
 
 	"github.com/mholt/caddy/caddytls"
 	// This is where other plugins get plugged in (imported)
+
```
Comment (elcore) [inline caddy/caddymain/run.go]: Please remove this!

Comment (juju2013) [inline caddy/caddymain/run.go]: oops, removed.

Comment (elcore) [inline caddy/caddymain/run.go]: Please, use/run gofmt https://golang.org/cmd/gofmt/

**Comment by elcore** [comment]
You still need to run gofmt -s

**Comment by juju2013** [comment]
gofmt -s produce the same output: 

gofmt -s run.go | sha1sum ; cat run.go | sha1sum
59ff1aab194355433995cef41bf043b12cf37857  -
59ff1aab194355433995cef41bf043b12cf37857  -

I'm missing something ?

**Comment by elcore** [comment]
You need to run it for caddyhttp/httpserver/plugin.go

**Comment by cez81** [comment]
`gofmt -w caddyhttp/httpserver/plugin.go`

You want the start of the comments to line up

**Comment by mholt** [comment]
This plugin, it only works if the entire label of a filename is all lowercase or all uppercase? Is this generally useful to lots of people?

**Comment by elcore** [comment]
Hello @juju2013 ,

any updates?

**Comment by tobya** [comment]
@juju2013 Thanks for the PR and Caddy plugin. 

Perhaps if rather than simply trying the lowercase and uppercase versions of the url, if you were to match with strings.EqualFold it would be a more general plugin.

**Comment by mholt** [comment]
Closing due to inactivity. @juju2013, If you decide to continue going forward on this plugin, consider this feedback first, answer the clarifying questions, and we'll be happy to help ensure your plugin gets published. :) Thanks!

**Pull request closed without merging**